smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -437,23 +453,105 @@ private void updateResponsesMapWithResponseStatusAndObject(
+        OpenApiConfig.ErrorStatusConflictHandlingStrategy strategy = context.getConfig().getOnErrorStatusConflict();
+        Map<String, List<MediaTypeObject>> contents = new LinkedHashMap<>();
+
+        // make response object for each error response & add to list
+        for (StructureShape shape : shapes) {
+            errors.add(createResponse(context, bindingIndex, eventStreamIndex, statusCode, shape,
+                    true));
+        }
+
+        // combine the error response objects into one response object
+        for (ResponseObject error : errors) {
+            // combine headers
+            for (Map.Entry<String, Ref<ParameterObject>> header : error.getHeaders().entrySet()) {
+                combinedResponse.putHeader(header.getKey(), header.getValue());
+            }
+
+            // get mapping of contents
+            for (Map.Entry<String, MediaTypeObject> content : error.getContent().entrySet()) {
+                contents.computeIfAbsent(content.getKey(), (k) -> new ArrayList<>()).add(content.getValue());
+            }
+
+            // combine links
+            for (Map.Entry<String, Ref<LinkObject>> link : error.getLinks().entrySet()) {
+                combinedResponse.putLink(link.getKey(), link.getValue());
+            }
+        }
+
+        // combine contents
+        for (Map.Entry<String, List<MediaTypeObject>> mapping : contents.entrySet()) {
+            if (mapping.getValue().size() == 1) {
+                combinedResponse.putContent(mapping.getKey(), mapping.getValue().get(0));
+            } else {
+                combinedResponse.putContent(mapping.getKey(), combineContents(
+                        strategy, mapping.getValue()));
+            }
+        }
+
+        // set description, build, and add to responses
+        combinedResponse.description("UnionError " + statusCode + " response");
+        responses.put(statusCode, combinedResponse.build());
+    }
+
+    private MediaTypeObject combineContents(OpenApiConfig.ErrorStatusConflictHandlingStrategy strategy,
+                                                         List<MediaTypeObject> contents) {
+        MediaTypeObject.Builder result = MediaTypeObject.builder();
+        Schema.Builder schema = Schema.builder();
+
+        if (strategy == OpenApiConfig.ErrorStatusConflictHandlingStrategy.USE_ONEOF) {
+            List<Schema> schemas = new ArrayList<>();
+            for (MediaTypeObject content : contents) {
+                schemas.add(content.getSchema().get());
+            }
+            schema.oneOf(schemas);
+        } else if (strategy == OpenApiConfig.ErrorStatusConflictHandlingStrategy.USE_PROPERTIES) {
+            int i = 1;
+            for (MediaTypeObject content : contents) {
+                schema.putProperty("Schema_" + i++, content.getSchema().get());
+            }
+        } else {
+            throw new OpenApiException("\"onErrorStatusConflict\" OpenAPI config setting has an invalid value.");
```

Comment:
The setting is valid by definition, i.e., is part of the enum, probably a better wording would be "unknown" (at least to this code), also probably better to use `IllegalArgumentException`, but definitely include the value that you cannot handle in the exception message,  e.g.,

```
                throw new IllegalArgumentException("Unknown strategy: " + strategy);
```